### PR TITLE
Remove unnecessary proto marshaler tests, avoid using InternalRep in otlprecv

### PR DESCRIPTION
**Pull request by bogdandrutu**
Updates https://github.com/open-telemetry/opentelemetry-collector/issues/3104

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: receiver/otlpreceiver/otlp_test.go [hunk 1]
```diff
@@ -240,74 +214,6 @@ func testHTTPJSONRequest(t *testing.T, url string, sink *internalconsumertest.Er
...
-	assert.NoError(t, err)
-
-	var resourceSpansOtlp2 otlptrace.ResourceSpans
-	err = jsonpb.UnmarshalString(json, &resourceSpansOtlp2)
-	assert.NoError(t, err)
-
-	assert.EqualValues(t, resourceSpansOtlp, resourceSpansOtlp2)
-}
-
-func TestJsonUnmarshaling(t *testing.T) {
```
Comment (tigrannajaryan) [inline receiver/otlpreceiver/otlp_test.go:258]: These are important tests that verify JSON marshaling. Why delete these?

**Review by tigrannajaryan — CHANGES_REQUESTED** [review]
Blocking since unclear why the tests are deleted.

File: receiver/otlpreceiver/otlp_test.go (hunk 1 above)
Comment (bogdandrutu) [reply receiver/otlpreceiver/otlp_test.go:258]: We are verifying that jsonpb which is from proto does the right thing... We should not unittest third party libraries

Comment (bogdandrutu) [reply receiver/otlpreceiver/otlp_test.go:258]: Friendly ping @tigrannajaryan

**Pull request merged**